Update dependencies

dobladov opened the pull request:
### Infrastructure
- Updated dependencies
- Upgraded from node `12` to `20`

dobladov committed:
Update minor and patch

dobladov committed:
Updates

dobladov committed:
Changelog

dobladov committed:
Changelog

dobladov committed:
Changelog

dobladov committed:
Merge remote-tracking branch 'origin/main' into Updates

dobladov committed:
Cleanup

dobladov committed:
use glob directly

dobladov committed:
Add types

dobladov commented:
@nitzanashi do you mind taking another review? I made some changes like adding TS to be more confident that the API's didn't change much after the updates.

nitzanashi commented on src/index.js:
```diff
@@ -1,24 +1,27 @@
-
-const fetchNPMVersions = (packageName, token) => {
-    return npmFetch.json(
+/**
+ * @param {string} packageName 
+ * @param {string} token
+ */
+const fetchNPMVersions = async (packageName, token) => {
```
Return value is missing

dobladov replied:
It's in the next line no?

nitzanashi replied:
Type wise I  mean `@returns`

dobladov replied:
I left it out because the inferred value was correct, or you are with ThePrimeagen on this and want all returns to be manually typed?

<img width="729" alt="Screenshot 2024-02-15 at 14 00 55" src="https://github.com/zattoo/publish/assets/1938043/18911b21-b500-411e-9d98-a9001176ab9f">

nitzanashi replied:
Na it's fine

nitzanashi reviewed (approved):
How are we going to test this?

dobladov committed:
Add propper testing

dobladov committed:
Add trigger

dobladov commented:
> How are we going to test this?

That's an excellent question, I was also wondering why this repo does not have any test.
I moved the functionality that was modified to utils and added some test, I really wanted an excuse to use node's internal test runner.

~I think the action can't run until it's merged into master.~

dobladov committed:
Remove space

dobladov commented:
@nitzanashi if you consider this package it's too critical, we can create a v2 for the release in case there's something wrong with the updates

nitzanashi reviewed (approved):
@dobladov All good, we can always revert back if needed, let's take a leap
But please communicate it to users, that we updated this service

nitzanashi commented on src/tests/utils.test.js:
```diff
@@ -0,0 +1,28 @@
+    getBody, 
+    fetchNPMVersions
+} = require  ('../utils.js');
+const { test } = require  ("node:test");
```
```suggestion
const { test } = require  ('node:test');
```

dobladov committed:
Update src/tests/utils.test.js

Co-authored-by: Nitzan <nitzan14@gmail.com>

dobladov commented:
@z-merge it
